### DISCUSSION TITLE
Dynamic routing client-side only

### DISCUSSION THREAD

**Discussion by underfisk** [vercel/next.js#15709] [Help] [answered]
Is there any way that next support Client-side routing because dynamic pages force the page to reload, most of the pages work fine but when we have a dynamic page it will force a page refresh
Currently, i have been using react-router, it serves the purpose but kinda makes Nextjs routing useless at some point

**Comment by ijjk** [comment, marked as answer]
Hi, it sounds like you might have incorrect `href` and `as` values on `next/link` causing the full page reload. The `href` should be the page in your pages folder and the `as` should be the URL shown. See more info on `href` and `as` for dynamic routes here in the docs https://nextjs.org/docs/api-reference/next/link#dynamic-routes

**Comment by underfisk** [comment]
@ijjk Okay it makes more sense now, kinda skipped that part because i didn't saw that the href was just for the file system and not the actual location
